src/server_manager/web_app/path_api.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as errors from '../infrastructure/errors';
+import {HttpRequest, HttpResponse} from '../electron_app/http/types';
```

Comment:
This is a circular dependency. The web app shouldn't depend on the electron app.

My suggestion:
- Move this file to `infrastructure/`. This is utility code not necessarily tied to our app. Imagine this as a third-party library, or a library that could be used by other apps.
- Define the request and response types here.
- Make the electron and web apps depend on the library in infrastructure.
- Create a library in the web app that defines the makePathApiClient that uses the `fetchWithPin` exposed method. Make the Server implementations depend on that.

Reply:
Done.  This required changing `rootDir` in `tsconfig.json` to allow the code in `electron_app/` to see the `infrastructure/` folder, and modifying the build action to account for the new `rootDir`.